[docs] Complete pickers customization pages (#8066)
mui/mui-x: docs/data/date-pickers/custom-components/custom-components.md

/>
```

> To modify components position, have a look at the [custom layout](/x/react-date-pickers/custom-layout/) docs page.

Reviewer: ```suggestion
To modify the component's position, have a look at the [custom layout](/x/react-date-pickers/custom-layout/) docs page.
```

## Tabs

The tabs are available on all date time picker components.

> It allows to switch between date and time interfaces.

Reviewer: ```suggestion
It allows switching between date and time interfaces.
```

{{"demo": "Tabs.js"}}

## Toolbar

The toolbar is available on all date time picker components.
It displays the current values and allows to switch between different views.

### Component props

> You can customize how the toolbar display the current value with `toolbarFormat`.

Reviewer: ```suggestion
You can customize how the toolbar displays the current value with `toolbarFormat`.
```

By default empty values are replaced by `__`.

> This can be modified by using `toolbarPlaceholder` props

Reviewer: ```suggestion
This can be modified by using `toolbarPlaceholder` props.
```

By default, the toolbar is `hidden` on desktop, and `visible` on mobile.
This behavior can be overridden by setting the `hidden` prop:

```jsx
<DatePicker
  slotProps={{
    toolbar: {
      // Customize value display
      toolbarFormat: 'YYYY',

>       toolbarPlaceholder: '??',

Reviewer: ```suggestion
      // Change what is displayed given an empty value
      toolbarPlaceholder: '??',
```

>       // Hidde the toolbar

Reviewer: ```suggestion
      // Show the toolbar
```
Or
`// Unhide the toolbar`

      hidden: false,
    },
  }}
/>
```

### Component

> Each component comes with its own toolbar (`DatePickerToolbar`, `DatePickerToolbar`, and `DateTimePickerToolbar`) that you can reuse and customize.

Reviewer: ```suggestion
Each component comes with its own toolbar (`DatePickerToolbar`, `TimePickerToolbar`, and `DateTimePickerToolbar`) that you can reuse and customize.
```

You can pass custom components—to replace the icons, for example—as shown below:

{{"demo": "ArrowSwitcherComponent.js", "defaultCodeOpen": false}}

## Shortcuts

> You can add shortcuts to every pickers.

Reviewer: ```suggestion
You can add shortcuts to every picker component.
```
